presto-orc/src/main/java/io/prestosql/orc/metadata/OrcMetadataReader.java

```diff
@@ -282,8 +282,19 @@ else if (statistics.hasBinaryStatistics()) {
             minAverageValueBytes = 0;
         }
 
+        Long numberOfValues = statistics.getNumberOfValues();
+        // To handle special case of Hive writer issue during minor compaction (HIVE-20604):
+        // After minor compaction, stripe stats don't have column statistics and bit field of numberOfValues
+        // is set to 1, but the value is wrongly set to default 0.
+        // If row count is 0, then set number of values to be null.
+        //
+        // NOTE: Presto orc writer doesn't write 'hasNull' field, added this check to skip in case of Presto.
+        if (statistics.hasHasNull() && statistics.getNumberOfValues() == 0 && !statistics.getHasNull()) {
+            numberOfValues = null;
```

Comment:
I am concerned this will affect Presto writer when we start to write `hasNull` -- there will be an unfair penalty for columns that are all-NULL within given stripe, but we drop this information.

1. we should drop the information only if the writer is the Hive writer
   * this information is indicated by https://github.com/prestosql/orc-protobuf/blob/af42e00911afac529bde22e7744011aabff7ea87/src/main/protobuf/orc_proto.proto#L354
  * it would need to be passed from `io.prestosql.orc.proto.OrcProto.Footer` to `io.prestosql.orc.metadata.Footer` in `io.prestosql.orc.metadata.OrcMetadataReader#readFooter`
2. we should not drop the information for new hive writer, where this is fixed
   * from my cursory reading of hive code, they did not change the writer version when they fixed this bug,
      but it seems they have changed the writer version later, in ORC-517, so we could avoid dropping this information for newer hives?

in fact we could recognize broken stats based on all columns information, not on column-by-column basis, this would reduce chances of dropping useful information.

Reply:
After more consideration, I think this code can stay as is -- if the number of values is 0 and we know there are no nulls, it's something wrong with the stats... we could drop them altogether with
```
return new ColumnStatistics(null, null, null, ....)
```

Please add a comment that it's a patalogical case, and then i think it's OK to remove a NOTE about Presto writer.

Reply:
Thank you for the feedback @findepi , I have rephrased the comment please take a look.